rust/numaflow-core/src/source/sqs.rs

```diff
@@ -12,13 +12,14 @@ impl TryFrom<SQSMessage> for Message {
     type Error = Error;
 
     fn try_from(message: SQSMessage) -> crate::Result<Self> {
-        let offset = Offset::String(StringOffset::new(message.offset, 0));
+        let offset = Offset::String(StringOffset::new(message.offset, *get_vertex_replica()));
         Ok(Message {
             keys: Arc::from(vec![message.key]),
             tags: None,
             value: message.payload,
-            offset: Some(offset.clone()),
+            offset: offset.clone(),
             event_time: message.event_time,
+            watermark: None,
```

Comment:
why not set `Watermark` based of `SentTimestamp` in the [Attributes]( https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_ReceiveMessage.html)